src/providers/cloudinary/runtime.ts

```diff
@@ -6,27 +6,45 @@ const operationsGenerator = createOperationsGenerator({
     fit: 'c',
     width: 'w',
     height: 'h',
-    format: 'f'
+    format: 'f',
+    quality: 'q'
   },
   valueMap: {
     fit: {
       fill: 'fill',
       inside: 'pad',
       outside: 'lpad',
       cover: 'fit',
-      contain: 'scale'
+      contain: 'scale',
+      minCover: 'mfit',
+      minInside: 'mpad',
+      thumbnail: 'thumb',
+      cropping: 'crop',
+      coverLimit: 'limit'
     }
   },
   joinWith: ',',
   formatter: (key, value) => `${key}_${value}`
 })
 
+const defaultModifiers = {
+  format: 'auto',
+  quality: 'auto'
+}
+
 export default <RuntimeProvider> {
   getImage (src: string, modifiers: ImageModifiers, options: any) {
-    const operations = operationsGenerator(modifiers)
+    const mergeModifiers = {
+      ...modifiers,
+      format: modifiers.format || defaultModifiers.format,
+      quality: modifiers.quality || defaultModifiers.quality
+    }
+
+    const srcWithoutExtension = src.replace(/\.[^/.]+$/, '')
```

Comment:
What does happens if two images uploaded on same path and different format?

Reply:
There are no two images uploaded on the same path
without overriding each other, their unique public id is different and
decided by the file name, not the extension. In other words, if you upload
an image called 'kitten.png', and then upload 'kitten.jpg', the second one
will replace the first one in Cloudinary if there is `overwritten` enabled.
Otherwise, the second one will be saved with different name, or a random
public id, or `kitten_1` maybe, depending on how you configure it. :)

Reply:
Good to know. Then it is safe :)